new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java

```diff
@@ -0,0 +1,371 @@
+            String.format("getPodCreationTimestamp failed with ApiException for pod %s in namespace %s",
+                adminServerPodName, domainNamespace));
+
+    // get the managed server pods original creation timestamps
+    logger.info("Getting managed server pods original creation timestamps");
+    List<String> managedServerPodOriginalTimestampList = new ArrayList<>();
+    for (int i = 1; i <= replicaCount; i++) {
+      final String managedServerPodName = managedServerPrefix + i;
+      managedServerPodOriginalTimestampList.add(
+          assertDoesNotThrow(() -> getPodCreationTimestamp(domainNamespace, "", managedServerPodName),
+              String.format("getPodCreationTimestamp failed with ApiException for pod %s in namespace %s",
+                  managedServerPodName, domainNamespace)));
+    }
+
+    // add the new server pod compute resources limits: cpu: 1, requests: cpu: 0.5
+    String cpuLimit = "1";
+    String cpuRequest = "0.5";
+
+    // verify if cpu limit was set then the new value should be different than the original value
+    if (limits.get("cpu") != null) {
+      assertNotEquals(limits.get("cpu").getNumber().toString(), cpuLimit,
+          String.format("server pod compute resources cpu limit is already set to %s, set cpu limit to "
+              + "a different value", cpuLimit));
+    }
+
+    // verify if cpu request was set then the new value should be different than the original value
+    if (requests.get("cpu") != null) {
+      assertNotEquals(requests.get("cpu").getNumber().toString(), cpuRequest,
+          String.format("server pod compute resources cpu request is already set to %s, set cpu request to "
+              + "a different value", cpuRequest));
+    }
+
+    // add/modify the server pod resources by patching the domain custom resource
+    assertTrue(addServerPodResources(cpuLimit, cpuRequest),
+        String.format("Failed to add server pod compute resources for domain {0} in namespace {1}",
+            domainUid, domainNamespace));
+
+    // verify the admin server pod was restarted
+    checkPodRestarted(domainUid, domainNamespace, adminServerPodName, adminPodOriginalTimestamp);
+
+    // check that the admin server pod is back to be ready
+    checkPodReady(adminServerPodName, domainUid, domainNamespace);
+
+    // verify the managed server pods were rolling restarted and back to be ready
+    Map<String, String> managedServerPodsWithTimeStamps = new HashMap<>();
+    for (int i = 1; i <= replicaCount; i++) {
+      String managedServerPodName = managedServerPrefix + i;
+      managedServerPodsWithTimeStamps.put(managedServerPodName, managedServerPodOriginalTimestampList.get(i - 1));
+    }
+    logger.info("Verifying rolling restart occurred for managed servers {0} in namespace {1}",
+        managedServerPodsWithTimeStamps.keySet(), domainNamespace);
+    verifyRollingRestartOccurred(managedServerPodsWithTimeStamps, 1, domainNamespace);
```

Comment:
This assertion will make sure that the pods - admin server, managed servers are restarted in order and comes to ready state.
So you don't need to check anything after doing addServerPodResources(cpuLimit, cpuRequest)
Please see the example in ItMiiConfigMapOverride test testAddMiiSystemResources on how to use this assertion.

Reply:
fixed